core/txpool/blobpool/blobpool.go

```diff
@@ -0,0 +1,1460 @@
+						if err := p.store.Delete(id); err != nil {
+							log.Error("Failed to delete dropped transaction", "id", id, "err", err)
+						}
+					}
+					break
+				}
+			}
+		}
+	}
+	log.Debug("Blobpool tip threshold updated", "tip", tip)
+	pooltipGague.Update(tip.Int64())
+	p.updateStorageMetrics()
+}
+
+// validateTx checks whether a transaction is valid according to the consensus
+// rules and adheres to some heuristic limits of the local node (price and size).
+func (p *BlobPool) validateTx(tx *types.Transaction, blobs []kzg4844.Blob, commits []kzg4844.Commitment, proofs []kzg4844.Proof) error {
+	// Ensure the transaction adheres to basic pool filters (type, size, tip) and
+	// consensus rules
+	baseOpts := &txpool.ValidationOptions{
+		Config:  p.chain.Config(),
+		Accept:  1 << types.BlobTxType,
+		MaxSize: txMaxSize,
+		MinTip:  p.gasTip.ToBig(),
+	}
+	if err := txpool.ValidateTransaction(tx, blobs, commits, proofs, p.head, p.signer, baseOpts); err != nil {
+		return err
+	}
+	// Ensure the transaction adheres to the stateful pool filters (nonce, balance)
+	stateOpts := &txpool.ValidationOptionsWithState{
+		State: p.state,
+
+		FirstNonceGap: func(addr common.Address) uint64 {
+			// Nonce gaps are not permitted in the blob pool, the first gap will
+			// be the next nonce shifted by however many transactions we already
+			// have pooled.
+			return p.state.GetNonce(addr) + uint64(len(p.index[addr]))
+		},
+		ExistingExpenditure: func(addr common.Address) *big.Int {
+			if spent := p.spent[addr]; spent != nil {
+				return spent.ToBig()
+			}
+			return new(big.Int)
+		},
+		ExistingCost: func(addr common.Address, nonce uint64) *big.Int {
+			next := p.state.GetNonce(addr)
+			if uint64(len(p.index[addr])) > nonce-next {
+				return p.index[addr][int(tx.Nonce()-next)].costCap.ToBig()
+			}
+			return nil
+		},
+	}
+	if err := txpool.ValidateTransactionWithState(tx, p.signer, stateOpts); err != nil {
+		return err
+	}
+	// If the transaction replaces an existing one, ensure that price bumps are
+	// adhered to.
+	var (
+		from, _ = p.signer.Sender(tx) // already validated above
+		next    = p.state.GetNonce(from)
+	)
+	if uint64(len(p.index[from])) > tx.Nonce()-next {
+		// Account can support the replacement, but the price bump must also be met
+		prev := p.index[from][int(tx.Nonce()-next)]
+		switch {
+		case tx.GasFeeCapIntCmp(prev.execFeeCap.ToBig()) <= 0:
+			return fmt.Errorf("%w: new tx gas fee cap %v <= %v queued", txpool.ErrReplaceUnderpriced, tx.GasFeeCap(), prev.execFeeCap)
+		case tx.GasTipCapIntCmp(prev.execTipCap.ToBig()) <= 0:
+			return fmt.Errorf("%w: new tx gas tip cap %v <= %v queued", txpool.ErrReplaceUnderpriced, tx.GasTipCap(), prev.execTipCap)
+		case tx.BlobGasFeeCapIntCmp(prev.blobFeeCap.ToBig()) <= 0:
+			return fmt.Errorf("%w: new tx blob gas fee cap %v <= %v queued", txpool.ErrReplaceUnderpriced, tx.BlobGasFeeCap(), prev.blobFeeCap)
+		}
+		var (
+			multiplier = uint256.NewInt(100 + p.config.PriceBump)
```

Comment:
```diff
diff --git a/core/txpool/blobpool/blobpool.go b/core/txpool/blobpool/blobpool.go
index 7a261fdc4..6da21745e 100644
--- a/core/txpool/blobpool/blobpool.go
+++ b/core/txpool/blobpool/blobpool.go
@@ -1006,10 +1006,11 @@ func (p *BlobPool) validateTx(tx *types.Transaction, blobs []kzg4844.Blob, commi
                }
                var (
                        multiplier = uint256.NewInt(100 + p.config.PriceBump)
+                       onehundred = uint256.NewInt(100)
 
-                       minGasFeeCap     = new(uint256.Int).Div(new(uint256.Int).Mul(multiplier, prev.execFeeCap), uint256.NewInt(100))
-                       minGasTipCap     = new(uint256.Int).Div(new(uint256.Int).Mul(multiplier, prev.execTipCap), uint256.NewInt(100))
-                       minBlobGasFeeCap = new(uint256.Int).Div(new(uint256.Int).Mul(multiplier, prev.blobFeeCap), uint256.NewInt(100))
+                       minGasFeeCap     = new(uint256.Int).Div(new(uint256.Int).Mul(multiplier, prev.execFeeCap), onehundred)
+                       minGasTipCap     = new(uint256.Int).Div(new(uint256.Int).Mul(multiplier, prev.execTipCap), onehundred)
+                       minBlobGasFeeCap = new(uint256.Int).Div(new(uint256.Int).Mul(multiplier, prev.blobFeeCap), onehundred)
                )
                switch {
                case tx.GasFeeCapIntCmp(minGasFeeCap.ToBig()) < 0:
```
Getting rid of allocations one by one :P

Reply:
:P LGT<

Reply:
Actually, no. You got rid of 2 allocations at once, so it's one more than one by one. You need to resubmit a proper fix